packages/flutter_driver/lib/src/extension.dart

```diff
@@ -96,6 +99,18 @@ class _FlutterDriverExtension {
   final Map<String, CommandDeserializerCallback> _commandDeserializers = <String, CommandDeserializerCallback>{};
   final Map<String, FinderConstructor> _finders = <String, FinderConstructor>{};
 
+  /// With [_frameSync] enabled, flutter_driver will wait to perform an action
+  /// until there are no pending frames in the app under test.
+  ///
+  /// This greatly reduces flakiness as it prevents flutter_driver from
+  /// interacting with an app that is currently in a transition. However,
+  /// in same cases, an app will never reach a state where there are no pending
+  /// frames (e.g. when there is a continuous animation on the screen). For
+  /// these cases, [_frameSync] needs to be disabled and its the responsibility
+  /// of the test writer to make sure that no flutter_driver actions are
+  /// performed while the app is in a transitional state.
```

Comment:
I trimmed down the documentation here a little bit since this is documented on the public side (driver.dart) again.